### DISCUSSION TITLE
Automatic conversion/localisation of units

### DISCUSSION THREAD

**Discussion by lod** [grafana/grafana#39647] [Dashboards, Panels & Visualizations]
I work with strange foreigners with strange foreign ways. They say "y'all" and measure temperature in degrees farenheit, other units like distance and heaviness are similarly weird.

And while the obvious solution is that they are wrong and should simply change to support my world view, sadly they outnumber me. What's worse, my device collection is simultaneously stubborn and inconsistent in their reporting of values in weird units.

This is a touch annoying but managable with Grafana, I can use maths to convert between them, my current goto is to plot two queries with two axis and try to hide one of them. This gives me a dual C/F axis and looks ok.

However Grafana knows the unit and can do so much better.

There should be a user unit override preference setting, probably a few because some people are weirdly inconsistent in mixing their units. This would allow me to have all the weird units automatically converted to something sensible, and similarly my colleagues could continue to have everything expressed in stones per furlong without me loosing my precious remaining hairs.

For backwards compatibility by default no preference would be set and no units would change.

**Comment by lod** [comment]
Related discussion of locale based comma/dot formatting https://github.com/grafana/grafana/issues/4252

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
